waspc/src/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -55,9 +66,35 @@ createRouterTemplateData wasp = RouterTemplateData
     , _pagesToImport = pages
     }
     where
-        routes = Wasp.getRoutes wasp
+        routes = map (createRouteTemplateData wasp) $ Wasp.getRoutes wasp
         pages = map createPageTemplateData $ Wasp.getPages wasp
 
+createRouteTemplateData :: Wasp -> Wasp.Route.Route -> RouteTemplateData
+createRouteTemplateData wasp route = RouteTemplateData
+    { _urlPath = Wasp.Route._urlPath route
+    , _targetComponent = determineRouteTargetComponent wasp route
+    }
+
+determineRouteTargetComponent :: Wasp -> Wasp.Route.Route -> String
+determineRouteTargetComponent wasp route =
+    maybe
+        targetPageName 
+        determineRouteTargetComponent'
+        (Wasp.Page._authRequired pageFromRoute)
+    where
+        targetPageName = Wasp.Route._targetPage route
+        -- NOTE(matija): if no page with the name specified in the route, head will fail.
+        pageFromRoute = head $ filter (((==) targetPageName) . Wasp.Page._name) (Wasp.getPages wasp)
+
+        -- | Applied if authRequired property is present.
+        determineRouteTargetComponent' :: Bool -> String
+        determineRouteTargetComponent' authRequired =
+            if authRequired 
+            -- TODO(matija): would be nicer if this function name wasn't hardcoded here.
+            then "createAuthRequiredPage(" ++ targetPageName ++ ")"
```

Comment:
How is it that we insert this HOC call directly in the route definition? Why not insert it in the component file? Or do we even have component file, maybe we don't? Maybe this is ok, I am just wondering -> I know before we had special Page files generated, but I am guess we don't create those anymore so it makes sense to do it here, right?

Reply:
yes, here I am taking directly a page component provided by user and I wrap it in createAuthRequiredPage. We could also generate new files in which HOC takes their page in, haven't thought of it before. Both ways sound good to me, but this one (one currently implemented) is simpler?